rclcpp/include/rclcpp/strategies/static_memory_strategy.hpp

```diff
@@ -27,40 +27,64 @@ namespace memory_strategies
 
 namespace static_memory_strategy
 {
-  struct ObjectPoolBounds
-  {
-  public:
-    size_t max_subscriptions_;
-    size_t max_services_;
-    size_t max_clients_;
-    size_t max_executables_;
-    size_t pool_size_;
-
-    ObjectPoolBounds(size_t subs = 10, size_t services = 10, size_t clients = 10,
-      size_t executables = 1, size_t pool = 1024)
-    : max_subscriptions_(subs), max_services_(services), max_clients_(clients), max_executables_(
-        executables), pool_size_(pool)
-    {}
-  };
+struct ObjectPoolBounds
+{
+public:
+  size_t max_subscriptions_;
+  size_t max_services_;
+  size_t max_clients_;
+  size_t max_executables_;
+  size_t max_guard_conditions_;
+  size_t pool_size_;
+
+  ObjectPoolBounds(size_t subs = 10, size_t services = 10, size_t clients = 10,
+    size_t executables = 1, size_t guard_conditions = 2, size_t pool = 1024)
```

Comment:
The only thing with setters is that you then have to consider what happens if someone calls a setter while a loan is out. Since we're using it we can assert we won't do that or we can throw an exception or something. But with the constructor arguments then there is no chance that this occurs since the destructor needs to be called before it can be constructed again.

Another option would be to have a factory object which can have setters for the settings and then when you're done with changing the settings you can call create which will return the object based on your factory's current settings. The object then has those settings until you delete it.